File: DTSparkCore/src/main/java/com/kevin/java/actions/Take_RDD.java
package com.kevin.java.actions;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;

import java.util.List;

/**
 * @author kevin
 * @version 1.0
 * @description     take是获取指定的数据
 * @createDate 2019/1/2
 */
public class Take_RDD {

    public static void main(String[] args) {

        // 1.创建SparkConf，设置作业名称和模式
        SparkConf conf = new SparkConf().setAppName("Take_RDD").setMaster("local");

        // 2.JavaSparkContext是Spark的唯一通道
        JavaSparkContext sc = new JavaSparkContext(conf);

        String file = "DTSparkCore\\src\\main\\resources\\test.txt";

        // 4.获取文件数据
        JavaRDD<String> liens = sc.textFile(file);

        // 5.获取rdd中指定的前几个数据，如获取rdd中前两个数据
        List<String> take = liens.take(2);

        // 6.遍历数据
        for (String s:take) {
            System.out.println(s);
        }

        // 7.关闭JavaSparkContext
        sc.close();
    }
}

File: dt-spark-core/src/main/scala/com/kevin/scala/actions/Take_RDD.scala
package com.kevin.scala.actions

import org.apache.spark.{SparkConf, SparkContext}

/**
  * take是获取指定的数据
  */
object Take_RDD {

  def main(args: Array[String]): Unit = {
    val file = "DTSparkCore\\src\\main\\resources\\test.txt"
    // 1.创建SparkConf
    val conf = new SparkConf().setAppName("Take_RDD").setMaster("local")
    // 2.创建SparkContext
    val sc = new SparkContext(conf)
    // 3.textFile读取文件数据,take获取前两行数据,foreach遍历,println打印
    sc.textFile(file).take(2).foreach(println(_))
    // 4.关闭sc
    sc.stop()

  }

}
